source/system-administrators/activities/configure-solrcloud-for-crafter.rst

```diff
@@ -0,0 +1,81 @@
+.. index:: Configure SolrCloud for Crafter; SolrCloud
+
+.. _configure-solrcloud-for-crafter:
+
+===============================
+Configure SolrCloud for Crafter
+===============================
+
+Crafter Search allows defining a CloudSolrClient in Spring config.  This section details how to configure SolrCloud for Crafter CMS.
+
+------------
+In Authoring
+------------
+#. Zip up (with tar.gz) the directory https://github.com/craftercms/search/tree/master/crafter-search-provider/solr/configsets/crafter_configs/conf.
+#. Copy the zip to one of the Solr servers and unzip it
+#. Go to **SOLR_INSTALL_DIR/server/scripts/cloud-scripts**.
+#. Run the command  ``./zkcli.sh -zkhost ZOOKEEPER_HOST:ZOOKEEPER_PORT -cmd upconfig -confname crafter_configs -confdir PATH_TO_UNZIPPED_CONFIGS`` (make sure the folder at PATH_TO_UNZIPPED_CONFIGS contains the ``solrconfig.xml`` file).
+#. Go to one of the Solr Admin Consoles.
+#. Go to **Collections** and click on **Add Collection**.
+#. Make the name of the collection the same as your site name, pick ``crafter_configs`` in config set and then click on **Add Collection**.
+#. Stop Authoring if it’s running.
+#. Go to **CRAFTER_INSTALL/bin/apache-tomcat/conf/Catalina/localhost** and create the file ``crafter-search-solrcloud.xml`` with the following content:
+
+   .. code-block:: xml
+
+       <?xml version="1.0" encoding="UTF-8"?>
+       <Context>
+           <Environment name="crafter.search.extension.base" value="classpath*:crafter/search-solrcloud/extension" type="java.lang.String" override="false"/>
+       </Context>
+
+   |
+
+#. Go to **CRAFTER_INSTALL/bin/apache-tomcat/webapps** and duplicate the ``crafter-search.war`` with the name ``crafter-search-solrcloud.war``.
+#. Go to **CRAFTER_INSTALL/bin/apache-tomcat/shared/classes/crafter**, create the folders **search-solrcloud/extension**, and inside create the file ``services-context.xml`` with the following content:
+
+   .. code-block:: xml
+       :linenos:
+
+       <?xml version="1.0" encoding="UTF-8"?>
+       <beans xmlns="http://www.springframework.org/schema/beans"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:context="http://www.springframework.org/schema/context"
+           xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd
+			       http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context.xsd">
+           <bean id="crafter.solrClient" class="org.craftercms.search.utils.spring.CloudSolrClientFactoryBean">
+                <property name="zkHost" value="ZOOKEEPER_HOST:ZOOKEEPER_PORT"/>
```

Comment:
I think we need to explain that people have to put the actual Zookeeper host and port.